searches/fibonacci_search.py

```diff
@@ -1,51 +1,100 @@
-# run using python fibonacci_search.py -v
-
 """
-@params
-arr: input array
-val: the value to be searched
-output: the index of element in the array or -1 if not found
-return 0 if input array is empty
+This is pure Python implementation of fibonacci search.
+
+Resources used:
+https://en.wikipedia.org/wiki/Fibonacci_search_technique
+
+For doctests run following command:
+python3 -m doctest -v fibonacci_search.py
+
+For manual testing run:
+python3 fibonacci_search.py
 """
+import sys
+from functools import lru_cache
+
+
+@lru_cache
+def fibonacci(k: int) -> int:
+    """Finds fibonacci number in index k.
 
+    Parameters
+    ----------
+    k : int
+        Index of fibonacci.
 
-def fibonacci_search(arr, val):
+    Returns
+    -------
+    int
+        Fibonacci number in position k.
 
+    >>> print(fibonacci(0))
+    0
+    >>> print(fibonacci(2))
+    1
+    >>> print(fibonacci(5))
+    5
+    >>> print(fibonacci(15))
+    610
     """
-    >>> fibonacci_search([1,6,7,0,0,0], 6)
+    if k == 0:
+        return 0
+    elif k == 1:
+        return 1
+    else:
+        return fibonacci(k - 1) + fibonacci(k - 2)
+
+
+def fibonacci_search(arr: list, val: int) -> int:
+    """A pure Python implementation of a fibonacci search algorithm.
+
+    Parameters
+    ----------
+    arr : list
+        List of sorted elements.
+    val : int
+        Element to search in list.
+
+    Returns
+    -------
+    int
+        The index of the element in the array.
+        -1 if the element is not found.
```

Comment:
```suggestion
    Parameters
    ----------
    arr : List of sorted elements.
    val : Element to search in the list.

    Returns
    -------
    The index of the element in the array.
    -1 if the element is not found.
```
As mentioned in the above comment.